gateleen-core/src/main/java/org/swisspush/gateleen/core/configuration/ConfigurationResourceManager.java

```diff
@@ -0,0 +1,232 @@
+        String resourceSchema = getRegisteredResources().get(requestUri);
+        storage.get(requestUri, buffer -> {
+            if (buffer != null) {
+                configurationResourceValidator.validateConfigurationResource(buffer, resourceSchema, event -> {
+                    if (event.succeeded()) {
+                        if (event.result().isSuccess()) {
+                            notifyObserversAboutResourceChange(requestUri, buffer.toString());
+                        } else {
+                            log.warn("Failure during validation of resource " + requestUri + ". Message: " + event.result().getMessage());
+                        }
+                    } else {
+                        log.warn("Failure during validation of resource " + requestUri + ". Message: " + event.cause());
+                    }
+                });
+            } else {
+                log.warn("Could not get URL '" + (requestUri == null ? "<null>" : requestUri) + "'.");
+            }
+        });
+    }
+
+    private void notifyObserverAboutResourceChange(String requestUri, ConfigurationResourceObserver observer) {
+        String resourceSchema = getRegisteredResources().get(requestUri);
+        storage.get(requestUri, buffer -> {
+            if (buffer != null) {
+                configurationResourceValidator.validateConfigurationResource(buffer, resourceSchema, event -> {
+                    if (event.succeeded()) {
+                        if (event.result().isSuccess()) {
+                            observer.resourceChanged(requestUri, buffer.toString());
+                        } else {
+                            log.warn("Failure during validation of resource " + requestUri + ". Message: " + event.result().getMessage());
+                        }
+                    } else {
+                        log.warn("Failure during validation of resource " + requestUri + ". Message: " + event.cause());
+                    }
+                });
+            } else {
+                log.warn("Could not get URL '" + (requestUri == null ? "<null>" : requestUri) + "'.");
+            }
+        });
+    }
+
+    private void notifyObserversAboutResourceChange(String resourceUri, String resource) {
+        List<ConfigurationResourceObserver> observersByResourceUri = getObserversByResourceUri(resourceUri);
+        for (ConfigurationResourceObserver observer : observersByResourceUri) {
+            observer.resourceChanged(resourceUri, resource);
+        }
+    }
+
+    public List<ConfigurationResourceObserver> getObserversByResourceUri(String resourceUri) {
+        List<ConfigurationResourceObserver> resourceObservers = getObservers().get(resourceUri);
+        if (resourceObservers == null) {
+            resourceObservers = new ArrayList<>();
+        }
+        return resourceObservers;
+    }
+
+    private enum ConfigurationResourceChangeType {
+        CHANGE, RESET;
```

Comment:
Is reset the right wording? As I understand the resource is getting deleted and the Observers have to informed about that. Perhaps just better documentation needed.